### ISSUE TITLE
After update user can login, but only gets error message

### ISSUE THREAD

**Issue by pekosro** [owncloud/core#19993]
### Steps to reproduce
1.Update OC 8.0.8 → 8.1.3
2.login as user (a user with shared files to already deleted users)
3.Error shown (no files shown) - "unknown error"

### Expected behaviour
Show normal file overview

### Actual behaviour
Only error message shown („unknown error“)

### Server configuration
**Operating system**:Ubuntu 14.04 LTS

**Web server:** apache2-bin (= 2.4.7-1ubuntu4.8)

**Database:** mysql-5.5

**PHP version:** php5 5.5.9+dfsg-1ubuntu4.13

**ownCloud version:** 8.1.3

**Updated from an older ownCloud or fresh install:**  update (from 8.0.8)

**List of activated apps:**
Enabled:
 - files: 1.1.10
 - files_external: 0.2.3
 - files_pdfviewer: 0.7
 - files_sharing: 0.6.2
 - files_texteditor: 0.4
 - files_trashbin: 0.6.3
 - files_videoviewer: 0.1.3
 - gallery: 0.6.1
 - updater: 0.6
 - user_ldap: 0.6.1
Disabled:                                                                                                                                                
 - activity                                                                                                                                                   
 - encryption                                                                                                                                              
 - external                                                                                                                                                    - files_locking                                                                                                                                            - files_versions                                                                                                                                           - firstrunwizard                                                                                                                                           - provisioning_api                                                                                                                                       - templateeditor                                                                                                                                          - user_external                                                                                                                                           - user_webdavauth                                                                                                                                                                                                                                                                                                 
**The content of config/config.php:**                                                                                                                         
<?php                                                                                                                                                         
$CONFIG = array (                                                                                                                                             
  'instanceid' => '51ac53cf97e8e',                                                                                                                            
  'passwordsalt' => 'XXXXXXXXXXXXXXXXXXXXXXXXXX',                                                                                                             
  'datadirectory' => '/var/www/ownclou /data',                                                                                                                
  'dbtype' => mysql',
  'version' => '8.1.3.0',                  
  'dbname' => 'owncloud',     
  'dbhost' => 'localhost',
  'dbtableprefix' => 'oc_',
  'dbuser' => 'admin',
  'dbpassword' => 'XXXXXX',
  'memcache.local' => '\\OC\\Memcache\\APCu',
  'installed' => true,
  'forcessl' => true,
  'ldapIgnoreNamingRules' => false,
  'loglevel' => '2',
  'maintenance' => false,
  'theme' => '',
  'mail_smtpmode' => 'smtp',
  'mail_smtphost' => 'smtp.bgc-jena.mpg.de',
  'mail_smtpport' => 25,
  'trusted_domains' => 
  array (
    0 => 'oc.bgc-jena.mpg.de',
    1 => 'oc.bgc.mpg.de',
    2 => 'oc2.bgc-jena.mpg.de',
    3 => 'tm.bgc-jena.mpg.de',
  ),
  'secret' => 'XXXXXXXXXX',
);

**Are you using external storage, if yes which one:** local/smb/sftp/...
NO
**Are you using encryption:** yes/no
NO
**Are you using an external user-backend, if yes which one:** 
LDAP

#### LDAP configuration (delete this part if not used)
+-------------------------------+-----------------------------------------------------------------------------------------+
| Configuration                 |                                                                                         |
+-------------------------------+-----------------------------------------------------------------------------------------+
| hasMemberOfFilterSupport      |                                                                                         |
| hasPagedResultSupport         |                                                                                         |
| homeFolderNamingRule          |                                                                                         |
| lastJpegPhotoLookup           | 0                                                                                       |
| ldapAgentName                 |                                                                                         |
| ldapAgentPassword             | ***                                                                                     |
| ldapAttributesForGroupSearch  |                                                                                         |
| ldapAttributesForUserSearch   |                                                                                         |
| ldapBackupHost                |                                                                                         |
| ldapBackupPort                |                                                                                         |
| ldapBase                      | dc=bgc-jena,dc=mpg,dc=de                                                                |
| ldapBaseGroups                | dc=bgc-jena,dc=mpg,dc=de                                                                |
| ldapBaseUsers                 | dc=bgc-jena,dc=mpg,dc=de                                                                |
| ldapCacheTTL                  | 600                                                                                     |
| ldapConfigurationActive       | 1                                                                                       |
| ldapEmailAttribute            | mail                                                                                    |
| ldapExperiencedAdmin          | 0                                                                                       |
| ldapExpertUUIDGroupAttr       |                                                                                         |
| ldapExpertUUIDUserAttr        |                                                                                         |
| ldapExpertUsernameAttr        |                                                                                         |
| ldapGroupDisplayName          | cn                                                                                      |
| ldapGroupFilter               | (&(|(objectclass=posixGroup)))                                                          |
| ldapGroupFilterGroups         |                                                                                         |
| ldapGroupFilterMode           | 0                                                                                       |
| ldapGroupFilterObjectclass    | posixGroup                                                                              |
| ldapGroupMemberAssocAttr      | memberUid                                                                               |
| ldapHost                      | ldapmaster.bgc-jena.mpg.de                                                              |
| ldapIgnoreNamingRules         |                                                                                         |
| ldapLoginFilter               | (&(|(objectclass=sambaSamAccount))(|(uid=%uid)(|(mailPrimaryAddress=%uid)(mail=%uid)))) |
| ldapLoginFilterAttributes     |                                                                                         |
| ldapLoginFilterEmail          | 1                                                                                       |
| ldapLoginFilterMode           | 0                                                                                       |
| ldapLoginFilterUsername       | 1                                                                                       |
| ldapNestedGroups              | 0                                                                                       |
| ldapNoCase                    | 0                                                                                       |
| ldapOverrideMainServer        | 0                                                                                       |
| ldapPagingSize                | 500                                                                                     |
| ldapPort                      | 389                                                                                     |
| ldapQuotaAttribute            |                                                                                         |
| ldapQuotaDefault              |                                                                                         |
| ldapTLS                       | 1                                                                                       |
| ldapUserDisplayName           | displayname                                                                             |
| ldapUserFilter                | (|(objectclass=sambaSamAccount))                                                        |
| ldapUserFilterGroups          |                                                                                         |
| ldapUserFilterMode            | 0                                                                                       |
| ldapUserFilterObjectclass     | sambaSamAccount                                                                         |
| ldapUuidGroupAttribute        | auto                                                                                    |
| ldapUuidUserAttribute         | auto                                                                                    |
| turnOffCertCheck              | 0                                                                                       |
| useMemberOfToDetectMembership | 1                                                                                       |
+-------------------------------+-----------------------------------------------------------------------------------------+


### Client configuration
**Browser:**
Firefox 41.0.1

**Operating system:**
opensuse 42.1

### Logs
#### Web server error log
- (no error)

#### ownCloud log (data/owncloud.log)
```
{"reqId":"Grm5wB6PYF4dEzXeb3iI","remoteAddr":"","app":"files","message":" Backends provided no user object for c0e294da-3d4a-1033-9459-874d129a0063","level":3,"time":"2015-10-22T14:01:52+00:00"}
{"reqId":"CVe5He6+xVKdIc\/EGkwJ","remoteAddr":"141.5.16.161","app":"files","message":" Backends provided no user object for c0e294da-3d4a-1033-9459-874d129a0063","level":3,"time":"2015-10-23T05:58:30+00:00"}
{"reqId":"CVe5He6+xVKdIc\/EGkwJ","remoteAddr":"141.5.16.161","app":"files","message":"Exception: {\"Exception\":\"OC\\\\User\\\\NoUserException\",\"Message\":\"Backends provided no user object for c0e294da-3d4a-1033-9459-874d129a0063\",\"Code\":0,\"Trace\":\"
#0 \\\/var\\\/www\\\/owncloud\\\/apps\\\/files_sharing\\\/lib\\\/cache.php(65): OC\\\\Files\\\\Filesystem::initMountPoints('c0e294da-3d4a-1...')\\n
#1 \\\/var\\\/www\\\/owncloud\\\/apps\\\/files_sharing\\\/lib\\\/cache.php(98): OC\\\\Files\\\\Cache\\\\Shared_Cache->getSourceCache('')\\n
#2 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/files\\\/view.php(1215): OC\\\\Files\\\\Cache\\\\Shared_Cache->get('')\\n
#3 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/files\\\/filesystem.php(861): OC\\\\Files\\\\View->getFileInfo('\\\/', true)\\n
#4 \\\/var\\\/www\\\/owncloud\\\/apps\\\/files\\\/ajax\\\/list.php(35): OC\\\\Files\\\\Filesystem::getFileInfo('\\\/')\\n
#5 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/route\\\/route.php(154) : runtime-created function(1): require_once('\\\/var\\\/www\\\/ownclo...')\\n
#6 [internal function]: __lambda_func(Array)\\n
#7 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/route\\\/router.php(274): call_user_func('?lambda_221', Array)\\n
#8 \\\/var\\\/www\\\/owncloud\\\/lib\\\/base.php(874): OC\\\\Route\\\\Router->match('\\\/apps\\\/files\\\/aja...')\\n
#9 \\\/var\\\/www\\\/owncloud\\\/index.php(40): OC::handleRequest()\\n
#10 {main}\",\"File\":\"\\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/files\\\/filesystem.php\",\"Line\":385}","level":4,"time":"2015-10-23T05:58:30+00:00"}
{"reqId":"Ud3AHrV2bsL3nsin5CnO","remoteAddr":"141.5.16.161","app":"files","message":" Backends provided no user object for c0e294da-3d4a-1033-9459-874d129a0063","level":3,"time":"2015-10-23T05:58:43+00:00"}
{"reqId":"Ud3AHrV2bsL3nsin5CnO","remoteAddr":"141.5.16.161","app":"index","message":"Exception: {\"Exception\":\"OC\\\\User\\\\NoUserException\",\"Message\":\"Backends provided no user object for c0e294da-3d4a-1033-9459-874d129a0063\",\"Code\":0,\"Trace\":\"
#0 \\\/var\\\/www\\\/owncloud\\\/apps\\\/files_sharing\\\/lib\\\/cache.php(65): OC\\\\Files\\\\Filesystem::initMountPoints('c0e294da-3d4a-1...')\\n
#1 \\\/var\\\/www\\\/owncloud\\\/apps\\\/files_sharing\\\/lib\\\/cache.php(205): OC\\\\Files\\\\Cache\\\\Shared_Cache->getSourceCache('')\\n
#2 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/files\\\/cache\\\/cache.php(243): OC\\\\Files\\\\Cache\\\\Shared_Cache->getId('')\\n
#3 \\\/var\\\/www\\\/owncloud\\\/apps\\\/files_sharing\\\/lib\\\/cache.php(327): OC\\\\Files\\\\Cache\\\\Cache->getFolderContents('')\\n
#4 [internal function]: OC\\\\Files\\\\Cache\\\\Shared_Cache->searchByMime('image')\\n
#5 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/files\\\/view.php(1499): call_user_func_array(Array, Array)\\n
#6 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/files\\\/view.php(1452): OC\\\\Files\\\\View->searchCommon('searchByMime', Array)\\n
#7 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/files\\\/filesystem.php(757): OC\\\\Files\\\\View->searchByMime('image')\\n
#8 \\\/var\\\/www\\\/owncloud\\\/lib\\\/public\\\/files.php(73): OC\\\\Files\\\\Filesystem::searchByMime('image')\\n
#9 \\\/var\\\/www\\\/owncloud\\\/apps\\\/gallery\\\/ajax\\\/getimages.php(53): OCP\\\\Files::searchByMime('image')\\n
#10 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/route\\\/route.php(154) : runtime-created function(1): require_once('\\\/var\\\/www\\\/ownclo...')\\n
#11 [internal function]: __lambda_func(Array)\\n
#12 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/route\\\/router.php(274): call_user_func('?lambda_27', Array)\\n
#13 \\\/var\\\/www\\\/owncloud\\\/lib\\\/base.php(874): OC\\\\Route\\\\Router->match('\\\/apps\\\/gallery\\\/a...')\\n
#14 \\\/var\\\/www\\\/owncloud\\\/index.php(40): OC::handleRequest()\\n
#15 {main}\",\"File\":\"\\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/files\\\/filesystem.php\",\"Line\":385}","level":4,"time":"2015-10-23T05:58:43+00:00"}
{"reqId":"dMJ0rBPa4OA6V4dfty3C","remoteAddr":"141.5.16.161","app":"files","message":" Backends provided no user object for c0e294da-3d4a-1033-9459-874d129a0063","level":3,"time":"2015-10-23T05:59:16+00:00"}
{"reqId":"dMJ0rBPa4OA6V4dfty3C","remoteAddr":"141.5.16.161","app":"files","message":"Exception: {\"Exception\":\"OC\\\\User\\\\NoUserException\",\"Message\":\"Backends provided no user object for c0e294da-3d4a-1033-9459-874d129a0063\",\"Code\":0,\"Trace\":\"
#0 \\\/var\\\/www\\\/owncloud\\\/apps\\\/files_sharing\\\/lib\\\/cache.php(65): OC\\\\Files\\\\Filesystem::initMountPoints('c0e294da-3d4a-1...')\\n
#1 \\\/var\\\/www\\\/owncloud\\\/apps\\\/files_sharing\\\/lib\\\/cache.php(98): OC\\\\Files\\\\Cache\\\\Shared_Cache->getSourceCache('')\\n
#2 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/files\\\/view.php(1215): OC\\\\Files\\\\Cache\\\\Shared_Cache->get('')\\n
#3 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/files\\\/filesystem.php(861): OC\\\\Files\\\\View->getFileInfo('\\\/', true)\\n
#4 \\\/var\\\/www\\\/owncloud\\\/apps\\\/files\\\/ajax\\\/list.php(35): OC\\\\Files\\\\Filesystem::getFileInfo('\\\/')\\n
#5 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/route\\\/route.php(154) : runtime-created function(1): require_once('\\\/var\\\/www\\\/ownclo...')\\n
#6 [internal function]: __lambda_func(Array)\\n
#7 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/route\\\/router.php(274): call_user_func('?lambda_396', Array)\\n
#8 \\\/var\\\/www\\\/owncloud\\\/lib\\\/base.php(874): OC\\\\Route\\\\Router->match('\\\/apps\\\/files\\\/aja...')\\n
#9 \\\/var\\\/www\\\/owncloud\\\/index.php(40): OC::handleRequest()\\n
#10 {main}\",\"File\":\"\\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/files\\\/filesystem.php\",\"Line\":385}","level":4,"time":"2015-10-23T05:59:16+00:00"}
{"reqId":"1dF7ta4qzSbyAoAr+Kr3","remoteAddr":"141.5.16.161","app":"no app in context","message":"Installation is in transit between the old Encryption (ownCloud <= 8.0)\n\t\t\tand the new encryption. Please enable the \"Default encryption module\"\n\t\t\tand run 'occ encryption:migrate'","level":2,"time":"2015-10-23T05:59:34+00:00"}
```
#### Browser log

**Comment by PVince81** [comment]
This seems to be relevant:
```
{"reqId":"1dF7ta4qzSbyAoAr+Kr3","remoteAddr":"141.5.16.161","app":"no app in context","message":"Installation is in transit between the old Encryption (ownCloud <= 8.0)\n\t\t\tand the new encryption. Please enable the \"Default encryption module\"\n\t\t\tand run 'occ encryption:migrate'","level":2,"time":"2015-10-23T05:59:34+00:00"}
```

@pekosro have you done the encryption migration ?

**Comment by pekosro** [comment]
On 26.10.2015 12:15, Vincent Petry wrote:
>
> This seems to be relevant:
>
> |{"reqId":"1dF7ta4qzSbyAoAr+Kr3","remoteAddr":"141.5.16.161","app":"no 
> app in context","message":"Installation is in transit between the old 
> Encryption (ownCloud <= 8.0)\n\t\t\tand the new encryption. Please 
> enable the \"Default encryption module\"\n\t\t\tand run 'occ 
> encryption:migrate'","level":2,"time":"2015-10-23T05:59:34+00:00"} |
>
> @pekosro <https://github.com/pekosro> have you done the encryption 
> migration ?
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/owncloud/core/issues/19993#issuecomment-151103487>.
>
We have not enabled encryption, but I I did all steps descriped in the docu.


-- 
Bye,
     Peer
________________________________________________________

Max-Planck-Institut für Biogeochemie
Dr. Peer-Joachim Koch
Hans-Knöll Str.10            Telefon: ++49 3641 57-6705
D-07745 Jena                 Telefax: ++49 3641 57-7705

**Comment by DeepDiver1975** [comment]
@MorrisJobke NoUserException

**Comment by MorrisJobke** [comment]
Related to https://github.com/owncloud/core/pull/19980 ?

**Comment by pekosro** [comment]
On 26.10.2015 15:34, Morris Jobke wrote:
>
> Related to #19980 <https://github.com/owncloud/core/pull/19980> ?
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/owncloud/core/issues/19993#issuecomment-151157575>.
>
YES - this looks like the situation here.

I tried to
Konsole output
1188  sudo -u www-data php /var/www/owncloud/occ  files:scan --all
1197  sudo -u www-data php /var/www/owncloud/occ  ldap:check-user 
"28f1be52-3b31-102c-96a3-1b4724d2340c"
1199  sudo -u www-data php /var/www/owncloud/occ  user:delete 
"28f1be52-3b31-102c-96a3-1b4724d2340c"

And after this weekend this error message disappeared! (I've changed any 
settings,
permissions or other files).

Is there any clean up code running (from cron) ?

-- 
Bye,
     Peer
________________________________________________________

Max-Planck-Institut für Biogeochemie
Dr. Peer-Joachim Koch
Hans-Knöll Str.10            Telefon: ++49 3641 57-6705
D-07745 Jena                 Telefax: ++49 3641 57-7705

**Comment by MorrisJobke** [comment]
> Is there any clean up code running (from cron) ?

Yes. Once a day - AFAIK. I also fixed the "receive share from deleted user" which will be in 8.2.1 and 8.1.4

I think then we can close this. Thanks for the report anyway.

**Comment by MorrisJobke** [comment]
Closed as duplicate of #19980
